api/db/services/document_service.py

```
@@ -1094,6 +1094,17 @@ def _sync_progress(cls, docs: list[dict]):
             try:
                 tsks = TaskService.query(doc_id=d["id"], order_by=Task.create_time)
                 if not tsks:
+                    # Document is RUNNING but has no tasks — executor likely crashed before
+                    # tasks were created. Reset to FAIL after a grace period so the frontend
+                    # stops polling endlessly.
+                    begin_at = d.get("process_begin_at")
+                    if (d.get("run") == TaskStatus.RUNNING.value and begin_at
+                            and (datetime.now() - begin_at).total_seconds() > 600):
```

Ruff (0.16.3)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)